cmd/rook/ceph/cleanup.go

```diff
@@ -58,21 +62,60 @@ func startCleanUp(cmd *cobra.Command, args []string) error {
 			logger.Infof("successfully cleaned up %q directory", cleanupDirPath)
 		}
 		// Remove all the mon directories.
-		monDirs, err := filepath.Glob(path.Join(dataDirHostPath, "mon-*"))
-		if err != nil {
-			return errors.Wrapf(err, "failed to clean up mon directories on dataDirHostPath %q", dataDirHostPath)
+		if monSecret == "" {
+			return errors.Errorf("failed to clean up mon directories in the dataDirHostPath %q due to empty monitor secret", dataDirHostPath)
 		}
-		if len(monDirs) > 0 {
-			for _, monDir := range monDirs {
+
+		return cleanMonDir()
+	}
+
+	return nil
+}
+
+func cleanMonDir() error {
+	monDirs, err := filepath.Glob(path.Join(dataDirHostPath, "mon-*"))
+	if err != nil {
+		return errors.Wrapf(err, "failed to find mon directories on the dataDirHostPath %q", dataDirHostPath)
+	}
+
+	if len(monDirs) > 0 {
```

Comment:
You don't need this check, the iteration will just not run if the length is 0, there is nothing to iterate over.

Reply:
updated. But still checking the length for appropriate logging.